Allow provide multiple ports with fixed subnets

ricolin opened the pull request:
openstack machine ports will allow to set multiple networks to controller and workers by provide extra_fixed_subnets in label of a magnum cluster.
extra subnets for openstack machine ports needs to be set in cluster label.
For example

```
openstack coe cluster template create   --image $(openstack image show ubuntu-2204-kube-${version}s -c id -f value)   --external-network public   --dns-nameserver 8.8.8.8   --master-lb-enabled   --master-flavor m1.medium   --flavor m1.medium   --network-driver calico   --docker-storage-driver overlay2   --coe kubernetes   --fixed-network private   --fixed-subnet private-subnet   --label kube_tag=${version}   --label fixed_subnet_cidr=192.168.2.0/24  \
--label extra_fixed_subnets='[{"name": "private-test-subnet2"}]' --keypair key   k8s-${version}-ports
```

okozachenko1203 commented on docs/user/labels.md:
```diff
@@ -298,6 +298,15 @@ is often accomplished by deploying a driver on each node.
+    Extra fixed subnets. This allow us to provide extra subnets for cluster.
+    For example `[{"id": "$extra_subnet_ID1"}, {"id": "$extra_subnet_ID2"}]`
+    This is not a override for the default fixed_subnet but a added-on list.
+    So we can have multiple subnets suuport (from same fixed network).
```
```suggestion
    So we can have multiple subnets support (from same fixed network).
```

okozachenko1203 reviewed (commented):
@ricolin 
- could you share the use case of this function? 
- i think pre-commit job fails because you made a PR from your personal repo.

ricolin commented:
> @ricolin
> 
> * could you share the use case of this function?
The use case is a Magnum cluster env with needs to add  63 shares to the Manila share network.
As there is a 24 share limit per share network.
current plan to allow multiple networks to be attached on each cluster node.

okozachenko1203 reviewed (commented):
@ricolin  i think this PR will not work for your use case.

![image](https://github.com/user-attachments/assets/81025199-6a2e-4805-9f33-4fc27cfc1ba1)
We cannot add more ipv4 subnets 

```
    Message:               error reconciling the Cluster topology: failed to create OpenStackCluster.infrastructure.cluster.x-k8s.io: FieldValueTooMany: spec.subnets: Too many: 3: must have at most 2 items FieldValueInvalid: <nil>: Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation
```

ricolin committed:
Allow multiple ports for OpenStack Machine

Signed-off-by: ricolin <rlin@vexxhost.com>

ricolin commented on docs/user/labels.md:
```diff
@@ -298,6 +298,15 @@ is often accomplished by deploying a driver on each node.
 
+* `extra_fixed_subnets`
+
+    Extra fixed subnets. This allow us to provide extra subnets for machines in cluster.
+    For example `[{"id": "$extra_subnet_ID1"}, {"id": "$extra_subnet_ID2"}]`
+    This is not a override for the default fixed_subnet but a added-on list.
+    So we can have multiple subnets support (from same fixed network).
```
@okozachenko1203  The format is demanded by https://github.com/openstack/magnum/blob/master/magnum/common/neutron.py#L123
Also it's actually a filter parameter so it can also be "name", etc. I can update the doc for more info.

ricolin replied:
Actually I forgot I changed the function reference on the way. docs fixed

ricolin committed:
fix docs

Signed-off-by: ricolin <rlin@vexxhost.com>

ricolin commented:
Here's the template 
```
root@instance:~/atmosphere# openstack coe cluster template show k8s-v1.27.8
+-----------------------+-------------------------------------------------------------------------------------------------------------------+
| Field                 | Value                                                                                                             |
+-----------------------+-------------------------------------------------------------------------------------------------------------------+
| insecure_registry     | -                                                                                                                 |
| labels                | {'kube_tag': 'v1.27.8', 'fixed_subnet_cidr': '192.168.2.0/24', 'extra_fixed_subnets': '["private-test-subnet2"]'} |
| updated_at            | -                                                                                                                 |
| floating_ip_enabled   | True                                                                                                              |
| fixed_subnet          | private-test-subnet                                                                                               |
| master_flavor_id      | m1.medium                                                                                                         |
| uuid                  | e285f5b1-085e-4ee8-8cfa-ff123c5add6b                                                                              |
| no_proxy              | -                                                                                                                 |
| https_proxy           | -                                                                                                                 |
| tls_disabled          | False                                                                                                             |
| keypair_id            | key                                                                                                               |
| public                | False                                                                                                             |
| http_proxy            | -                                                                                                                 |
| docker_volume_size    | -                                                                                                                 |
| server_type           | vm                                                                                                                |
| external_network_id   | public                                                                                                            |
| cluster_distro        | ubuntu                                                                                                            |
| image_id              | e13400c7-a5d4-4d67-b648-26c94cbaecbd                                                                              |
| volume_driver         | -                                                                                                                 |
| registry_enabled      | False                                                                                                             |
| docker_storage_driver | overlay2                                                                                                          |
| apiserver_port        | -                                                                                                                 |
| name                  | k8s-v1.27.8                                                                                                       |
| created_at            | 2025-05-14T07:32:20+00:00                                                                                         |
| network_driver        | calico                                                                                                            |
| fixed_network         | private-test                                                                                                      |
| coe                   | kubernetes                                                                                                        |
| flavor_id             | m1.medium                                                                                                         |
| master_lb_enabled     | True                                                                                                              |
| dns_nameserver        | 8.8.8.8                                                                                                           |
| project_id            | a78b2608b5d0498b9b6c5d884dd1957b                                                                                  |
| hidden                | False                                                                                                             |
| tags                  | -                                                                                                                 |
+-----------------------+-------------------------------------------------------------------------------------------------------------------+

ricolin committed:
Bound extra_fixed_subnets with fixed_subnet

Raise ClusterInvalidLabel if extra_fixed_subnets label exist but
not cluster.fixed_subnet.

Signed-off-by: ricolin <rlin@vexxhost.com>

ricolin commented:
I added a checker in front so now changed to `requires fixed_subnet as mandatory if this label is set`
with failed reason:
``` 
status_reason        | Invalid Cluster label extra_fixed_subnets: Label `extra_fixed_subnets` requires cluster fixed subnet.
```
